Lombiq.HelpfulLibraries/Libraries/Utilities/RandomNumberGeneratorExtensions.cs

```diff
@@ -0,0 +1,42 @@
+namespace System.Security.Cryptography
+{
+    public static class RandomNumberGeneratorExtensions
+    {
+        /// <summary>
+        /// Returns a non-negative cryptographically secure random integer that is within a specified range.
+        /// </summary>
+        /// <param name="minValue">The inclusive lower bound of the random number returned.</param>
+        /// <param name="maxValue">
+        /// The exclusive upper bound of the random number returned. <paramref name="maxValue"/> must be greater than or
+        /// equal to <paramref name="minValue"/>.
+        /// </param>
+        /// <returns>A cryptographically random number within the specified range.</returns>
+        /// <remarks>
+        /// <para>
+        /// Taken from
+        /// <see href="https://docs.microsoft.com/en-us/archive/msdn-magazine/2007/september/net-matters-tales-from-the-cryptorandom"/>.
+        /// </para>
+        /// </remarks>
+        public static int Next(this RandomNumberGenerator randomNumberGenerator, int minValue, int maxValue)
+        {
+            if (minValue > maxValue) throw new ArgumentOutOfRangeException(nameof(minValue));
+            if (minValue == maxValue) return minValue;
+
+            long diff = maxValue - minValue;

```

Comment:
You should cast at least one of the operands to `long` first otherwise you still risk overflow/underflow becuase the result of the operation is evaluated as `int` first and converted to the `long` only at assignment.
![image](https://user-images.githubusercontent.com/4007293/158011845-2fa8f9d9-d121-4771-b1f3-089a5722691a.png)
As an added bonus if you cast like this then you can keep using `var` too.

Reply:
Good catch!